packages/components/Tabs/src/TabsItem.tsx

```diff
@@ -39,24 +38,32 @@ export const TabsItem = compose<TabsItemType>({
     // Grabs the context information from Tabs (currently selected TabsItem and client's onTabsClick callback).
     const info = React.useContext(TabsContext);
 
-    /* There's a bug where the user callback is being called multiple times on one click.
-    We check that there is an actual change in selection before forwarding the message to the user callback
-    so that they aren't notified more than once for each click. */
-    const changeSelection = () => {
-      if (itemKey != info.selectedKey) {
+    const [focusState, setFocusState] = React.useState({
+      focused: false,
+    });
+
+    const changeSelection = React.useCallback(() => {
+      componentRef?.current?.focus();
+    }, [componentRef]);
+
+    const changeSelectionWithFocus = React.useCallback(() => {
+      setFocusState({ focused: true });
+      if (componentRef.current && !focusState.focused) {
```

Comment:
Why do we need to check for `componentRef.current` here? componentRef is an optional prop, correct? If the user doesn't pass one in, this will never change selection?

Reply:
It is an optional prop, but if the user doesn't pass one, componentRef is set to defaultComponentRef. This can be seen on line 24 and 29 of TabsItem.tsx.

Reply:
What does `componentRef.current` equal when the user doesn't pass one in?

Reply:
Wouldn't it be null? And then `componentRef.current` wouldn't be true?

Reply:
If the user does not pass one, it would be equal to defaultComponentRef = React.useRef(null).
And it was evaluating for true as the functions bellow where being called.

Reply:
Changed it so only ```!focusState.focused``` gets evaluated on this if statement.